assets/js/googlesitekit/widgets/datastore/areas.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * core/widgets data store: widgets info.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Store our widget components by registry, then by widget `slug`. We do this because
+ * we can't store React components in our data store.
+ *
+ * @private
+ * @since n.e.x.t
+ */
+export const WidgetComponents = {};
+
+const ASSIGN_WIDGET_AREA = 'ASSIGN_WIDGET_AREA';
+const REGISTER_WIDGET_AREA = 'REGISTER_WIDGET_AREA';
+
+export const INITIAL_STATE = {
+	areas: {},
+	areaWidgets: {},
```

Comment:
Can we name these more according to our actual defined names:
* `areas` --> `widgetAreas`
* `areaWidgets` --> `widgets`
The current names you used for these properties are rather confusing.

**Update:** Actually `areaWidgets` shouldn't be here at all, since that storage should be (and is) part of `widgets.js`.